shared/metrics/types.go

```diff
@@ -0,0 +1,171 @@
+package metrics
```

Comment:
This should go in `lxd/metrics`, `shared/` is for public facing packages that are used by our CLI tools or external integrations. If something is only used by lxd and lxd-agent, it should go under lxd/